crates/telio-traversal/src/endpoint_providers/stun.rs

```diff
@@ -604,6 +671,30 @@ impl<Wg: WireGuard, E: Backoff> State<Wg, E> {
             })
             .unwrap_or(false)
     }
+
+    fn ipv6_is_enabled(&self) -> bool {
+        if let Some(sockets) = self.sockets.as_ref() {
+            return sockets.tun_socket_v6.is_some();
+        }
+
+        false
+    }
+
+    fn should_fallback_to_ipv4(&self) -> bool {
+        self.ipv6_is_enabled() && self.current_proto == IpProto::IPv6
+    }
+
+    fn get_socket(&self) -> Option<Arc<UdpSocket>> {
+        if let Some(sockets) = self.sockets.as_ref() {
+            if self.current_proto == IpProto::IPv6 {
+                sockets.tun_socket_v6.clone()
```

Comment:
please add here `debug_assert!` to verify that `ipv6_is_enabled()`.

Reply:
I've added `is_in_ipv6_mode` check. What's the point of `debug_assert!` here?

Reply:
That is fine, no need for `debug_assert` in that case.